addons/sale_project/models/sale_order.py

```diff
@@ -88,56 +105,88 @@ def _action_confirm(self):
 
     def action_view_task(self):
         self.ensure_one()
+        if not self.order_line:
+            return {'type': 'ir.actions.act_window_close'}
 
         list_view_id = self.env.ref('project.view_task_tree2').id
         form_view_id = self.env.ref('project.view_task_form2').id
+        kanban_view_id = self.env.ref('project.view_task_kanban_inherit_view_default_project').id
 
-        action = {'type': 'ir.actions.act_window_close'}
-        task_projects = self.tasks_ids.mapped('project_id')
-        if len(task_projects) == 1 and len(self.tasks_ids) > 1:  # redirect to task of the project (with kanban stage, ...)
-            action = self.with_context(active_id=task_projects.id).env['ir.actions.actions']._for_xml_id(
-                'project.act_project_project_2_project_task_all')
-            action['domain'] = [('id', 'in', self.tasks_ids.ids)]
-            if action.get('context'):
-                eval_context = self.env['ir.actions.actions']._get_eval_context()
-                eval_context.update({'active_id': task_projects.id})
-                action_context = safe_eval(action['context'], eval_context)
-                action_context.update(eval_context)
-                action['context'] = action_context
-        else:
-            action = self.env["ir.actions.actions"]._for_xml_id("project.action_view_task")
-            action['context'] = {}  # erase default context to avoid default filter
-            if len(self.tasks_ids) > 1:  # cross project kanban task
-                action['views'] = [[False, 'kanban'], [list_view_id, 'tree'], [form_view_id, 'form'], [False, 'graph'], [False, 'calendar'], [False, 'pivot']]
-            elif len(self.tasks_ids) == 1:  # single task -> form view
-                action['views'] = [(form_view_id, 'form')]
-                action['res_id'] = self.tasks_ids.id
+        default_line = next(sol for sol in self.order_line if sol.product_id.detailed_type == 'service')
+
+        action = self.env["ir.actions.actions"]._for_xml_id("project.action_view_task")
+        action['context'] = {}  # erase default context to avoid default filter
```

Comment:
?

Reply:
yes, it was asked in the spec, if we let the default context, there's gonna be the default filter 'my task' which is not needed here. Since we're opening all the task from the SO, it made little sense to group them with it.

Reply:
It is not anymore indeed, since we update the context later, we can just set the context on L124 instead of updating the empty dict we're creating there.